openhtf/output/web_gui/__init__.py

```diff
@@ -331,9 +332,37 @@ def __init__(self, discovery_interval_s, disable_discovery, http_port,
         (r'/', self.MainHandler, {'port': http_port}),
         (r'/station/(?:(?:[0-9]{1,3}\.){3}[0-9]{1,3})/(?:[0-9]{1,5})(?:/*)',
          self.MainHandler, {'port': http_port}),
-        (r'/(.*\..*)', tornado.web.StaticFileHandler, {'path': path}),
-        (r'/(styles\.css)', tornado.web.StaticFileHandler, {'path': path}),
+        (r'/(.*\..*)', tornado.web.StaticFileHandler, {'path': path})
     ] + dash_router.urls + station_router.urls
+
+    #Add plugins into handler_routes
+    if type(plugins) == list:
+      for name in plugins:
+        try:
+          #Import plugin package
+          plugin_module = importlib.import_module(name)
+          handler,deps = plugin_module.handler
+
+          #Add in handler dependencies 
+          dep_dict = {}
+          for dep in deps:
+            if dep == 'station_store':
+              dep_dict['station_store'] = self.store
+            elif dep == 'host':
+              dep_dict['host'] = socket.gethostname()
+            elif dep == 'port':
+              dep_dict['port'] = http_port
+            elif dep == 'path':
+              dep_dict['path'] = path
+            else: 
+              print("Unable to find dependancy "+dep)
+
+          #Add completed route to handler_routes
+          handler.append(dep_dict)
+          handler_routes.append(tuple(handler))
+        except:
```

Comment:
* Put the try-except around the minimal amount of code you [expect to fail]. It looks like you should only put it around the first two lines (`import_module` and `plugin_module.handler` may fail)
* Catch more specific exceptions, such as `ImportError` and `AttributeError`
* Use _LOG.exception to log your reasoning for the error as well as the traceback itself

[expect to fail]: https://google.github.io/styleguide/pyguide.html?showone=Exceptions#Exceptions